python/tank/platform/engine.py

```diff
@@ -628,6 +629,59 @@ def threaded_work(val):
             # we don't have an invoker so just call the function:
             return func(*args, **kwargs)
 
+    def get_matching_commands(self, command_selectors):
+        """
+        Finds all the commands that match the given selectors.
+
+        Note that selectors that do not match a command will output a warning.
+
+        :param command_selectors: A list of command selectors, with each
+                                  selector having the following structure:
+                                    {
+                                      name: command-name,
+                                      app_instance: instance-name
+                                    }
+                                  An empty name ('') will select all the
+                                  commands of the given instance-name.
+
+        :returns:                 A list of tuples for all commands that match
+                                  the selectors. Each tuple has the format:
+                                    (instance-name, command-name, callback)
+        """
+        # return a dictionary grouping all the commands by instance name
+        commands_by_instance = {}
+        for (name, value) in self.commands.iteritems():
+            app_instance = value["properties"].get("app")
+            if app_instance is None:
+                continue
+            instance_name = app_instance.instance_name
+            commands_by_instance.setdefault(instance_name, []).append(
+                (name, value["callback"]))
+
+        # go through the selectors and return any matching commands
+        ret_value = []
+        for selector in command_selectors:
+            command_name = selector["name"]
+            instance_name = selector["app_instance"]
+            instance_commands = commands_by_instance.get(instance_name, [])
+
+            # add the commands if the name of the settings is ''
+            # or the name matches
+            matching_commands = [(instance_name, name, callback)
+                                 for (name, callback) in instance_commands
+                                 if not command_name or (command_name == name)]
+            ret_value.extend(matching_commands)
+
+            # give feedback if no commands were found
+            if not matching_commands:
+                self._engine.log_warning(
+                    "The requested command '%s' from app '%s' could not be "
+                    "matched.\nPlease make sure that you have the app "
+                    "installed and that it is properly loaded." %
```

Comment:
Minor suggestion: "and that it has successfully initialized" instead of "and that it is properly loaded"?